engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobUpdateTimeoutTest.java

```diff
@@ -197,4 +198,67 @@ private static void assertJobDeadline(
             ENGINE.getClock().getCurrentTimeInMillis() + timeout,
             within(Duration.ofMillis(100).toMillis()));
   }
+
+  @Test
+  public void shouldUpdateJobTimeoutForCustomTenant() {
+    // given
+    final String tenantId = "acme";
+    ENGINE.createJob(jobType, PROCESS_ID, Collections.emptyMap(), tenantId);
+    final var batchRecord =
+        ENGINE
+            .jobs()
+            .withType(jobType)
+            .withTimeout(Duration.ofMinutes(5).toMillis())
+            .withTenantId(tenantId)
+            .activate();
+
+    final JobRecordValue job = batchRecord.getValue().getJobs().get(0);
+    final long jobKey = batchRecord.getValue().getJobKeys().get(0);
+    final long timeout = Duration.ofMinutes(10).toMillis();
+
+    // when
+    final Record<JobRecordValue> updatedRecord =
+        ENGINE
+            .job()
+            .withKey(jobKey)
+            .withTimeout(timeout)
+            .withAuthorizedTenantIds(tenantId)
+            .updateTimeout();
+
+    // then
+    Assertions.assertThat(updatedRecord.getValue()).hasTenantId(tenantId);
+    assertJobDeadline(updatedRecord, jobKey, job, timeout);
+  }
+
+  @Test
+  public void shouldRejectUpdateRetriesIfTenantIsUnauthorized() {
+    // given
+    final String tenantId = "acme";
+    final String falseTenantId = "foo";
+    ENGINE.createJob(jobType, PROCESS_ID, Collections.emptyMap(), tenantId);
+    final var batchRecord =
+        ENGINE
+            .jobs()
+            .withType(jobType)
+            .withTimeout(Duration.ofMinutes(5).toMillis())
+            .withTenantId(tenantId)
+            .activate();
+
+    final JobRecordValue job = batchRecord.getValue().getJobs().get(0);
+    final long jobKey = batchRecord.getValue().getJobKeys().get(0);
+    final long timeout = Duration.ofMinutes(10).toMillis();
+
+    // when
+    final Record<JobRecordValue> jobRecord =
+        ENGINE
+            .job()
+            .withKey(jobKey)
+            .withTimeout(timeout)
+            .withAuthorizedTenantIds(falseTenantId)
+            .expectRejection()
+            .updateTimeout();
+
+    // then
+    Assertions.assertThat(jobRecord).hasRejectionType(RejectionType.NOT_FOUND);
+  }
```

Comment:
🔧 To improve readability and consistency, the new test methods could move above the helper method.

This is not blocking, it's just a matter of style. I'd suggest to improve this while we're at it.

Reply:
Absolutely agree, thank you.
This file is marked as `Outdated` and I've checked that the new test methods are above the helper method.

Reply:
Funny, the last commit I can see now wasn't there when I reviewed this. Strange, well, maybe the page was cached. It looks good, we can resolve this here 👍